tests/ui/traits/next-solver/typeck/normalize-in-upvar-collection.rs

```diff
@@ -0,0 +1,15 @@
+//@ compile-flags: -Znext-solver
+//@ check-pass
+
```

Comment:
pls add comment linking to this PR and maybe mention that it's a minimization of `icu_provider_adaptors`